src/tpm.rs

```diff
@@ -525,6 +591,54 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_get_tpm_metadata_1() {
+        match set_tpmdata_test() {
+            Ok(_) => {
+                // using test tpmdata.json content must present, system won't panic
+                let remove: &[_] = &['"', ' ', '/'];
+                let password =
+                    get_tpm_metadata_content("aik_handle").unwrap();
+                assert_eq!(
+                    password.unwrap().trim_matches(remove),
+                    String::from("FB1F19E0")
+                );
+            }
+            Err(code) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_get_tpm_metadata_2() {
+        match set_tpmdata_test() {
+            Ok(_) => match get_tpm_metadata_content("foo") {
+                Ok(content) => assert_eq!(content, None),
+                Err(e) => assert!(false),
+            },
+            Err(code) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_write_tpm_metadata() {
+        match set_tpmdata_test() {
+            Ok(_) => match set_tpm_metadata_content("owner_pw", "hello") {
+                Ok(_) => {
+                    // using test tpmdata.json content must present, system won't panic
+                    let remove: &[_] = &['"', ' ', '/'];
+                    let password =
+                        get_tpm_metadata_content("owner_pw").unwrap();
+                    assert_eq!(
+                        password.unwrap().trim_matches(remove),
+                        String::from("hello")
+                    )
+                }
+                Err(_) => assert!(false),
```

Comment:
Instead of this, I would do something like: `assert!(set_tpmdata_test().is_ok())`

Reply:
This is great.  Changed.